client/components/seo-preview-pane/index.jsx

```diff
@@ -24,27 +27,67 @@ import {
 	getSelectedSite
 } from 'state/ui/selectors';
 
-const largeBlavatar = site => `${ get( site, 'icon.img', '//gravatar.com/avatar/' ) }?s=512`;
+const PREVIEW_IMAGE_WIDTH = 512;
+const largeBlavatar = site => `${ get( site, 'icon.img', '//gravatar.com/avatar/' ) }?s=${ PREVIEW_IMAGE_WIDTH }`;
+
+const getPostImage = ( post ) => {
+	if ( ! post ) {
+		return null;
+	}
+
+	// Use the featured image if one was set
+	if ( post.featured_image ) {
+		return post.featured_image;
+	}
+
+	// Otherwise we'll look for a large enough image in the post
+	const content = post.content;
+	if ( ! content ) {
+		return null;
+	}
+
+	const imgElements = parseHtml( content ).querySelectorAll( 'img' );
+	const largeImage = get( find( imgElements, ( { width } ) => width >= PREVIEW_IMAGE_WIDTH ), 'src', null );
```

Comment:
if this gets confusing, we can extract the `find()` into its own variable, as much as I'm not a fan of it

sometimes I also rewrite the arguments on their own line to visually separate what's happening…

```js
const largeImage = get(
	find( imgElements, ( { width } ) => width >= PREVIEW_IMAGE_WIDTH ),
	'src',
	null
);
```